packages/grid/_modules_/grid/hooks/utils/useLogger.ts

```diff
@@ -49,19 +50,21 @@ let factory: LoggerFactoryFn | null;
 
 export function useLoggerFactory(
   apiRef: any,
-  { logger, logLevel }: { logger?: Logger; logLevel?: string | false },
+  props: Pick<GridComponentProps, 'logger' | 'logLevel'>,
```

Comment:
Maybe we can have it as a rule if we all agree - if there are more than 2 then have it as `props`?

```suggestion
 { logger, logLevel }: Pick<GridComponentProps, 'logger' | 'logLevel'>,
```

Reply:
https://github.com/mui-org/material-ui-x/pull/2275#discussion_r682585408

I agree that we should agree on a common rule :+1: